tsconfig.json

```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "lib": ["es2018", "dom"],
+    "lib": ["es2018", "dom", "esnext"],
```

Comment:
@dmisdm added this at the top level. Not sure if it's worth having a package-level jest config? Let me know

Reply:
Let's have a package level config for this - it might introduce issues for downstream users of this package if we switch everything over to `esnext`.

Reply:
So when I add it to the package level like so:
```
{
  "extends": "../../tsconfig.dist.json",
  "compilerOptions": {
    "allowJs": false,
    "declaration": true,
    "noEmit": false,
    "sourceMap": true,
    "lib": ["es2018", "dom", "esnext"]
  },
  "exclude": ["docs"]
}
```

I get this error:
```
➜  kaizen-design-system git:(ls/testBrandMoment) ✗ yarn test BrandMoment
yarn run v1.22.11
$ jest BrandMoment
 FAIL  packages/brand-moment/src/BrandMoment.spec.tsx
  ● Test suite failed to run

    node_modules/@cultureamp/unified-navigation-ui/lib/helpers/uiHelpers.ts:126:13 - error TS2339: Property 'flatMap' does not exist on type 'MenuItem[]'.

    126   menuItems.flatMap((item, idx) => {
                                 ~~~~~~~
```

Reply:
@ActuallyACat bump on this

Reply:
@ActuallyACat just got a stale notification so opening this back up. Do you have any way to solve the config issue above?

Reply:
Sorry - let me look into this. Kaizen supports many different consumers with different needs, so we need to exercise some caution as using esnext features might cause issues for our users.

Reply:
Gutfeel - is it worth adding `@cultureamp/unified-navigation-ui` to this repo?

Reply:
Unsure! What would that solve?

Reply:
The issue is `unified-navigation-ui` is using `flatMap` - a function that is not supported in `es2018`. If we remove it the issue is avoided - which is a bandaid solution. I'm pretty hesitant to use `esnext` because there's no guarantee our users will correctly polyfill all `esnext` features (that, and esnext it's a moving target). 

If we bump to `es2019` that will successfully compile. I'm going to push a commit and chase up what's the minimum target environment for our users.

Reply:
I removed support for private packages recently, as no one was actively using them in this repo. Now the build is broken because of that change. ಠ_ಠ

Reply:
Removed unified-navigation. The component doesn't specify what shape of React node it expects.